_posts/2020-10-22-how-to-update-to-aws-provider-v3.adoc

```diff
@@ -0,0 +1,191 @@
+---
+title: How to update Gruntwork references to Version 3 of the Terraform AWS provider
+categories: Foundations
+image: /assets/img/guides/infrastructure-as-code-library/grunty-grid.png
+excerpt: Learn how to update the Gruntwork Reference Architecture and your infrastructure code that depends on the Gruntwork Infrastructure as Code Library to use the latest version of the Terraform AWS provider.
+tags: ["aws", "terraform", "terragrunt"]
+cloud: ["aws"]
+redirect_from: /static/guides/foundations/how-to-update-to-aws-provider-v3/
+---
+:page-type: guide
+:page-layout: post
+
+:toc:
+:toc-placement!:
+
+// GitHub specific settings. See https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74 for details.
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+toc::[]
+endif::[]
+
+== Intro
+
+This guide will walk you through how to update the https://gruntwork.io/reference-architecture/[Gruntwork Reference
+Architecture] and any code that depends on the
+https://gruntwork.io/infrastructure-as-code-library/[Gruntwork Infrastructure as Code Library] to the latest version
+(v3) of the Terraform AWS provider.
```

Comment:
```suggestion
https://gruntwork.io/infrastructure-as-code-library/[Gruntwork Infrastructure as Code Library] to the version 3.x of the Terraform AWS provider.
```

It won't be "latest" forever.

Reply:
Also, consider adding a sentence here as to _why_ this matters: e.g., version 3.x has lots of new features (link), but also a number of backwards incompatible changes that have to be incorporated carefully into your codebase.

Reply:
Well, version 3.x doesn't really have a single page I can link to with "new features" (there are 12 minor releases, each with "new" stuff)... The real thing here is that 2.x is deprecated and nothing new will be going in there.

This is why I went for the following phrasing in the section below:

> This means that if you wish to use any new resources or data sources that are released from this point onwards, you will need to be compatible with version 3 of the AWS provider to be able to pull those changes in.

I'll copy a similar statement to the intro.

Reply:
Done in 6ea3831db7e56d9bb73ff6e6d0697e500690f24d